app/angular/src/server/config.js

```diff
@@ -47,22 +48,28 @@ export default function(configType, baseConfig, configDir) {
     logger.info('=> Loading angular-cli config.');
   }
 
+  const defaultConfig = applyAngularCliWebpackConfig(
+    createDefaultWebpackConfig(config),
+    cliWebpackConfigOptions
+  );
+
   // Check whether user has a custom webpack config file and
   // return the (extended) base configuration if it's not available.
   const customConfigPath = path.resolve(configDir, 'webpack.config.js');
 
   if (!fs.existsSync(customConfigPath)) {
     logger.info('=> Using default webpack setup based on "angular-cli".');
-    const configPath = path.resolve(__dirname, './config/defaults/webpack.config.js');
-    const customConfig = require(configPath);
-
-    return applyAngularCliWebpackConfig(customConfig(config), cliWebpackConfigOptions);
+    return defaultConfig;
   }
   const customConfig = require(customConfigPath);
 
   if (typeof customConfig === 'function') {
     logger.info('=> Loading custom webpack config (full-control mode).');
-    return customConfig(applyAngularCliWebpackConfig(config, cliWebpackConfigOptions), configType);
+    return customConfig(
+      applyAngularCliWebpackConfig(config, cliWebpackConfigOptions),
+      configType,
+      defaultConfig
```

Comment:
So in case of full control mode, there is a first parameter that is extended with angular-cli's config, and the default config which is extended as well. Is this expected?

Reply:
Given that you pick one or the other, yes